middleware/checkFilter.js

```diff
@@ -0,0 +1,19 @@
+async function checkFilter(req, res, next) {
+    const filter = ["reviewCount", "reviewAvg", "answerCount"];
+    let validFilter = false;
+
+    filter.forEach((element) => {
+        if (element === req.params.filter) {
+            validFilter = true;
+        }
+    });
+
+    if (!validFilter) {
+        return res.status(400).json({ message: "잘못된 필터 입니다." });
+    }
```

Comment:
```suggestion
    if(!filter.includes(req.params.filter)){
        return res.status(400).json({ message: "잘못된 필터 입니다." });
    }
    next();
```

그리고 sort로 변경하기로 하지 않았나요??